### PR TITLE
Make sure each Compositor::IDisplay::Instance() call is Released()

### PR DESCRIPTION
**Pull request by VeithMetro**

### CODE REVIEW

**Review by sebaszm — APPROVED** [review]

File: src/thunder/renderer-backend.cpp [hunk 1]
```diff
@@ -140,11 +140,25 @@ struct wpe_renderer_backend_egl_interface thunder_renderer_backend_egl_interface
     // destroy
     [](void* data)
     {
+        if (data != nullptr) {
+            Thunder::Compositor::IDisplay* display = static_cast<Thunder::Compositor::IDisplay*>(data);
```
Comment (pwielders) [inline src/thunder/renderer-backend.cpp:144]: This is against the rules of reference counting. You get the interface offered, so the on offering you this interface has to have a Reference count. You do not maintain the interface and thus correctly do not do an AddRef (or QueryInterface) on this interface and thus you should not Release it as now you are releasing on beha;f of someone else...

File: src/thunder/renderer-backend.cpp [hunk 2]
```diff
@@ -140,11 +140,25 @@ struct wpe_renderer_backend_egl_interface thunder_renderer_backend_egl_interface
...
+            display->Release();
+        }
     },
     // get_native_display
     [](void* data) -> EGLNativeDisplayType
     {
-        return Thunder::Compositor::IDisplay::Instance(Thunder::DisplayName())->Native();
```
Comment (pwielders) [inline src/thunder/renderer-backend.cpp:147]: I think that this was your leak, that you where trying to solve above, cause the method Thunder::Compositor::IDisplay::Instance(Thunder::DisplayName()) returns a reference counted interface to the callee and the callee is now responsible to Release. By now using the data, the release is no needed, see discussion above..

**Review by pwielders — CHANGES_REQUESTED** [review]
It is causing to many Releases now or it is fixed in the wrong spot..

**Review by pwielders — APPROVED** [review]
Love it!

**Pull request merged**